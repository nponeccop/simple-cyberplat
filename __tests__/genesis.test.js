const axios = require('axios')
const fsp = require('fs').promises
const { Genesis } = require('../index')

jest.mock('axios')

/** Genesis setup and request  */
async function requestBoilerplate (requestBody) {
  const key = await fsp.readFile('testKeys/privateKey.pem', 'utf-8')
  return await new Genesis({
    defaultFields: {
      AP: 1111
    },
    provider: {
      url: 'https://predproc.tothemars.io/pay'
    },
    crypto: {
      key,
      passphrase: ''
    }
  }).request(requestBody)
}

afterEach(() => {
  jest.restoreAllMocks()
})

test('utf8-win1251 transcoding', async () => {
  axios.request.mockImplementationOnce(async (axiosConf) => {
    expect(axiosConf).toMatchSnapshot()
    // rawResponse is a Windows 1251 BLOB/Buffer, not a string! Thus a file
    const rawResponse = await fsp.readFile('__tests__/testData/response1.txt')
    return axiosConf.transformResponse(rawResponse)
  })
  const result = await requestBoilerplate({
    REQ_TYPE: 'мама мыла раму'
  })
  expect(axios.request).toHaveBeenCalledTimes(1)
  expect(result).toMatchSnapshot()
})
