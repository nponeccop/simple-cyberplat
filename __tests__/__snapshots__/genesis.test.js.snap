// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utf8-win1251 transcoding 1`] = `
Object {
  "data": Object {
    "data": Array [
      65,
      80,
      61,
      49,
      49,
      49,
      49,
      13,
      10,
      82,
      69,
      81,
      95,
      84,
      89,
      80,
      69,
      61,
      236,
      224,
      236,
      224,
      32,
      236,
      251,
      235,
      224,
      32,
      240,
      224,
      236,
      243,
    ],
    "type": "Buffer",
  },
  "headers": Object {
    "Content-Type": "text/plain; encoding=windows-1251",
    "x-sign": "H5JyXfWP2ebpE8mIur5aO09nP4d64xV8bP6g7NpP87R5%2BkSiR9wU8s6%2FxghqvmHRukB5II2jGDYGAU5wQeXZyC0ZVj13sXP2MsdfY2Oy%2BhJ1T%2F4Q99LAZNl49ymZlbvHq%2B50D9jnOqZ07AsxOU34yTB%2BajzipoRKhBfW7V9OE30%3D",
  },
  "method": "POST",
  "responseType": "arraybuffer",
  "transformResponse": [Function],
  "url": "https://predproc.tothemars.io/pay",
}
`;

exports[`utf8-win1251 transcoding 2`] = `
Object {
  "BAR": "зайчики заплакали",
  "FOO": "уточки закрякали",
}
`;
